sympy/core/power.py

```diff
@@ -832,9 +832,10 @@ def _check(ct1, ct2, old):
                         as_int(pow, strict=False)
                         combines = True
                     except ValueError:
-                        combines = isinstance(Pow._eval_power(
-                            Pow(*old.as_base_exp(), evaluate=False),
-                            pow), (Pow, exp, Symbol))
+                        b, e = old.as_base_exp()
+                        # These conditions ensure that (b**e)**f == b**(e*f) for any f
+                        combines = b.is_positive and e.is_real or b.is_nonnegative and e.is_nonnegative
```

Comment:
`(b**e)**f` = b**(e*f)` also if `f.is_integer and (b.is_positive or e.is_integer)`. I think it was me that put this `combines` there and I wanted to make sure that we didn't hard-code in a test like this which would miss valid combinations or misapply invalid combinations.  (It took a long time to track down and remove those hard-coded applications down.) I don't know if @asmeurer note [here](https://github.com/sympy/sympy/pull/83#issuecomment-670411) has ever had a good reference point to consult.

Reply:
I positively view the addition of a function that validates a combing of exponents in the power.py file -- one that doesn't require evaluation of a power (which, as you note, has brought troubles of its own).

Reply:
In general I think that substitutions like `x.subs(x**2, y)` are not a good idea for subs. It seems like subs is used for several different things that I would have made separate functions for e.g.:
```
(1 + x).subs(x, 1) # replace a symbol with a "value"
sin(exp(x)).subs(exp(x), y) # replace a subexpression with a different subexpression
(x**4).subs(x**2, y) # attempt a basic elimination of x from the equation
```
I think it would have been better if `subs` only did the first of these things: replace a symbol with a value. For the second I think something like replace is better. For the third case there should just be a different function called e.g. `eliminate`. The third case is very different from the others because it ends up replacing things that are not literally in the expression tree e.g. there is no `Pow(x, 2)` in the expression tree for `Pow(x, 4)`.

Given that I (personally) think that using subs for elimination is a mistake I'm not sure how comprehensive the checking here should be.

Reply:
My intention for this PR was for the change to be as minimal as possible though. I would like to apply this to the 1.9 release branch but I don't want to do that if it looks like it would potentially break things.

Reply:
Although there is a failure of the symengine job (#22054) it looks like all other tests have passed. If you can think of any examples that would fail under this PR then it would be good to add them to the tests and fix them here.

Reply:
> not a good idea for subs

This has been my sentiment from old. We called these "algebraic substitutions". 

I can perhaps look more closely tonight or tomorrow.

Reply:
I think I see the problem: the conclusion was being based on the structure of the result (imagining that the result collapsed to a symbol or single exponential function or a power with combined exponent). But consider
```
>>> eq=x**Rational(3,2)
>>> eq**Rational(3,4)
(x**(3/2))**(3/4)

```
The last power does not represent successful combination of the exponents. So perhaps `_combines` should have been:
```python
combines = Pow._eval_power(
                            Pow(*old.as_base_exp(), evaluate=False),
                            pow).as_base_exp()[0] == old.as_base_exp()[0]
```